plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1737,6 +1756,17 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkSupportedDmlWriterVersion(ConnectorSession session, DeltaLakeTableHandle table)
+    {
+        SchemaTableName schemaTableName = table.getSchemaTableName();
+        int requiredWriterVersion = getProtocolEntry(session, schemaTableName).getMinWriterVersion();
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(table.getMetadataEntry());
+        if (requiredWriterVersion == MAX_DML_WRITER_VERSION && columnMappingMode != ColumnMappingMode.UNKNOWN) {
```

Comment:
We should be able to support any writer version.
What we don't support is writing into tables having features not currently supported by Trino (e.g. : tables with generated columns, tables with check constraints, ...)

Reply:
please check with https://github.com/trinodb/trino/issues/12638 what is required to full support of writer 5 version  (this PR is cover only first 2 items), so we cannot talk about full support of writer 5 version within context of that PR, since here about DML operations only. Elimination of this check and upgrade of MAX_WRITER_VERSION to 5 should be done with separated PR when all writer 5 requirements be fulfilled.